homeassistant/components/ruckus_unleashed/coordinator.py

```diff
@@ -37,12 +31,19 @@ def __init__(self, hass: HomeAssistant, *, ruckus: Ruckus) -> None:
 
     async def _fetch_clients(self) -> dict:
         """Fetch clients from the API and format them."""
-        clients = await self.ruckus.current_active_clients()
-        return {e[API_MAC]: e for e in clients[API_CURRENT_ACTIVE_CLIENTS][API_CLIENTS]}
+        try:
+            clients = await self.ruckus.api.get_active_clients()
+        except Exception as error:
+            _LOGGER.exception("Error getting clients: %s", error)
+            raise error
+        _LOGGER.debug("fetched %d active clients", len(clients))
+        return {client[API_CLIENT_MAC]: client for client in clients}
 
     async def _async_update_data(self) -> dict:
         """Fetch Ruckus Unleashed data."""
         try:
-            return {API_CLIENTS: await self._fetch_clients()}
-        except (AuthenticationError, ConnectionError) as error:
+            return {KEY_SYS_CLIENTS: await self._fetch_clients()}
+        except AuthenticationError as autherror:
+            raise UpdateFailed(autherror) from autherror
+        except Exception as error:
```

Comment:
We don't allow catching the bare `Exception` (except in the config_flow). Please refactor the code to catch only exception fired by the library.

Reply:
refactored